src/java/org/apache/cassandra/db/DeletionTime.java

```diff
@@ -187,8 +244,29 @@ public void skip(DataInputPlus in) throws IOException
 
         public long serializedSize(DeletionTime delTime)
         {
-            return TypeSizes.sizeof(delTime.localDeletionTime())
-                 + TypeSizes.sizeof(delTime.markedForDeleteAt());
+            return TypeSizes.sizeof(Integer.MAX_VALUE)
+                   + TypeSizes.sizeof(delTime.markedForDeleteAt());
+        }
+    }
+
+    // When scrubbing legacy sstables (overflown) or upon sstable corruption we could have negative ldts
+    public static class InvalidDeletionTime extends DeletionTime
+    {
+        public InvalidDeletionTime(long markedForDeleteAt, int localDeletionTime)
```

Comment:
We should probably make the constructor `private`, since this is meant to be called only by `DeletionTime.getInstance`.

Reply:
Done